app/editor/src/features/admin/report-templates/constants/defaultContentRazorTemplate.ts

```diff
@@ -0,0 +1,124 @@
+export const defaultContentSubjectRazorTemplate = `@using TNO.TemplateEngine
+@Settings.Subject.Text @(Settings.Subject.ShowTodaysDate ? $" - {ReportExtensions.GetTodaysDate():dd-MMM-yyyy}" : "")`;
+
+export const defaultContentBodyRazorTemplate = `@inherits RazorEngineCore.RazorEngineTemplateBase<TNO.TemplateEngine.Models.Reports.ReportEngineContentModel>
+@using System
+@using System.Linq
+@using TNO.Entities
+@using TNO.TemplateEngine
+@{
+  var pageBreak = Settings.Sections.UsePageBreaks ? "page-break-after: always;" : "";
+  var utcOffset = ReportExtensions.GetUtcOffset(System.DateTime.Now, "Pacific Standard Time");
+}
+<h2 id="top">Report Title</h2>
+<div style="color:red;">DO NOT FORWARD THIS EMAIL TO ANYONE</div>
+<br/>
+
+@if (Content.Count() == 0)
+{
+  <p>There is no content in this report.</p>
+}
+@if (Settings.ViewOnWebOnly)
+{
+  <a href="">Click to view this report online</>
+}
+else
+{
+  foreach (var section in Sections)
+  {
+    var sectionContent = section.Value.Content.ToArray();
+    if (section.Value.IsEnabled && sectionContent.Length > 0 || !Settings.Sections.HideEmpty)
+    {
+      <div style="@pageBreak">
+        @if (!String.IsNullOrEmpty(section.Value.Settings.Label))
+          <h3>@section.Value.Settings.Label</h3>
+         @if (!String.IsNullOrEmpty(section.Value.Description))
+          <p>@section.Value.Description</p>
+
+        @* Charts *@
+        @if (section.Value.Settings.ChartsOnTop && section.Value.Settings.ShowCharts)
+        {
+          foreach (var chart in section.Value.ChartTemplates)
+          {
+            <img alt="@chart.SectionSettings.AltText" src="@chart.Uid" />
+          }
+        }
+
+        @* Table of Contents *@
+        @if (section.Value.Settings.ShowContent)
+        {
+          <ul>
+            @foreach (var content in sectionContent)
+            {
+              var headline = $"{(Settings.Headline.ShowSentiment ? content.GetSentimentIcon("{0} - ") : "")}{content.Headline}{(Settings.Headline.ShowShortName && !String.IsNullOrEmpty(content.Source?.ShortName) ? $" - {content.Source?.ShortName}" : (Settings.Headline.ShowSource ? $" - {content.OtherSource}": ""))}{(Settings.Headline.ShowPublishedOn ? $" - {content.PublishedOn?.AddHours(utcOffset):dd-MMM-yyyy}" : "")}";
```

Comment:
There is a lot happening in this one line.  Would an Editor every need to understand this?

Reply:
Editors do not have access to any of this.  Only Admins can see this.  However, it would be rare to have an admin who would know how to edit these templates.  These are Razor templates which we store in the database.  I've added these default constants to speed up development.